src/python/pants/core/util_rules/system_binaries.py

```diff
@@ -506,7 +506,16 @@ async def find_binary(
         bash = await Get(BashBinary)
         shebang = f"#!{bash.path}"
 
-    script_path = "./find_binary.sh"
+    # HACK: For workspace environments, the `find_binary.sh` will be mounted in the "chroot" directory
+    # which is not the current directory (since the process will execute in the workspace). Thus,
+    # adjust the path used as argv[0] to find the script.
+    script_name = "find_binary.sh"
+    script_exec_path = (
+        f"./{script_name}"
+        if not env_target.is_workspace_environment()
+        else f"{{chroot}}/{script_name}"
+    )
```

Comment:
Is there a better way to handle this subtle difference between workspace environments and every other kind of environment? Remote environments do not support the `{chroot}` replacement so simply using `{chroot}/find_binary.sh` as the path won't work for all environments.

Reply:
I've seen worse hacks, but maybe instead of inspecting `is_workspace_environment()` there should be an `env_target.binary_root_path(chroot: str)` or some more general name (this is technically not just for binaries but for any sandboxed input).

Reply:
Introduced a `EnvironmentTarget.sandbox_base_path` method.